third-party/gmp/Makefile

```diff
@@ -28,7 +28,7 @@ CHPL_GMP_CFG_OPTIONS += --enable-static --disable-shared
 endif
 
 CHPL_GMP_ABI_ARG =
-ifeq (linux32, $(CHPL_MAKE_TARGET_PLATFORM))
+ifeq ($(CHPL_MAKE_TARGET_PLATFORM), $(filter $(CHPL_MAKE_TARGET_PLATFORM),linux32 cygwin32))
```

Comment:
I don't think we encounter this much, but someone could set CHPL_TARGET_PLATFORM=linux32 and compile from a 64bit system. In that case, I think we could want to look at CHPL_TARGET_PLATFORM.